cloud/pkg/edgecontroller/controller/downstream.go

```diff
@@ -438,6 +442,117 @@ func (dc *DownstreamController) syncEndpoints() {
 		}
 	}
 }
+func (dc *DownstreamController) syncPersistentvolumes() {
+	var operation string
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop edgecontroller downstream syncPersistentvolumes loop")
+			return
+		case e := <-dc.persistentvolumsManager.Events():
+			pv, ok := e.Object.(*v1.PersistentVolume)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", pv)
+				continue
+			}
+
+			switch e.Type {
+			case watch.Added:
+				dc.lc.AddOrUpdatePersistentVolume(*pv)
+				operation = model.InsertOperation
+			case watch.Modified:
+				dc.lc.AddOrUpdatePersistentVolume(*pv)
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				dc.lc.DeletePersistentVolume(*pv)
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("persistentVolume event type: %s unsupported", e.Type)
+				continue
+			}
+			// send to all nodes
+			dc.lc.EdgeNodes.Range(func(key interface{}, value interface{}) bool {
+				nodeName, ok := key.(string)
+				if !ok {
+					klog.Warning("Failed to assert key to sting")
+					return true
+				}
+				msg := model.NewMessage("")
+				msg.SetResourceVersion(pv.ResourceVersion)
+				resource, err := messagelayer.BuildResource(nodeName, pv.Namespace, common.ResourceTypePersistentVolume, pv.Name)
+				if err != nil {
+					klog.Warningf("Built message resource failed with error: %v", err)
+					return true
+				}
+				msg.BuildRouter(constants.EdgeControllerModuleName, constants.GroupResource, resource, operation)
+				msg.Content = pv
+				if err := dc.messageLayer.Send(*msg); err != nil {
+					klog.Warningf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
+				} else {
+					klog.V(4).Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
+				}
+				return true
+			})
+		}
+	}
+}
+
+func (dc *DownstreamController) syncPersistentvolumclaims() {
+	var operation string
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop edgecontroller downstream syncPersistentvolumclaims loop")
+			return
+		case e := <-dc.persistentvolumsManager.Events():
+			pvc, ok := e.Object.(*v1.PersistentVolumeClaim)
+			if !ok {
+				klog.Warningf("Object type: %T unsupported", pvc)
+				continue
+			}
+
+			switch e.Type {
+			case watch.Added:
+				dc.lc.AddOrUpdatePersistentvolumeclaim(*pvc)
+				operation = model.InsertOperation
+			case watch.Modified:
+				dc.lc.AddOrUpdatePersistentvolumeclaim(*pvc)
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				dc.lc.DeletePersistentvolumeclaim(*pvc)
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("Persistentvolumclaim event type: %s unsupported", e.Type)
+				continue
+			}
+			// send to all nodes
```

Comment:
Do we need to send to all nodes? Just sends to the node in localCache.

Reply:
> Do we need to send to all nodes? Just sends to the node in localCache.

get it,I have updated it. 
pvcmap and pvmap key is namespace/name ,value is nodename list.
 add pod and update pod sync pvcmap to cache. 
 add pvc and update pvc  sync pvmap to cache.